test/cppsuite/tests/background_compact.cpp

```diff
@@ -0,0 +1,328 @@
+        testutil_assert(collection_count != 0);
+        uint64_t collections_per_thread = collection_count / tw->thread_count;
+        /* Must have unique collections for each thread. */
+        testutil_assert(collection_count % tw->thread_count == 0);
+        for (int i = tw->id * collections_per_thread;
+             i < (tw->id * collections_per_thread) + collections_per_thread && tw->running(); ++i) {
+            collection &coll = tw->db.get_collection(i);
+            scoped_cursor cursor = tw->session.open_scoped_cursor(coll.name);
+            ccv.push_back({coll, std::move(cursor)});
+        }
+
+        uint64_t counter = 0;
+        while (tw->running()) {
+            if (maintenance_window) {
+                std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+                continue;
+            }
+
+            uint64_t start_key = ccv[counter].coll.get_key_count();
+            uint64_t added_count = 0;
+            tw->txn.begin();
+
+            /* Collection cursor. */
+            auto &cc = ccv[counter];
+            while (tw->txn.active() && tw->running()) {
+                /* Insert a key value pair, rolling back the transaction if required. */
+                auto key = tw->pad_string(std::to_string(start_key + added_count), tw->key_size);
+                auto value =
+                  random_generator::instance().generate_pseudo_random_string(tw->value_size);
+                if (!tw->insert(cc.cursor, cc.coll.id, key, value)) {
+                    added_count = 0;
+                    tw->txn.rollback();
+                } else {
+                    added_count++;
+                    if (tw->txn.can_commit()) {
+                        if (tw->txn.commit()) {
+                            /*
+                             * We need to inform the database model that we've added these keys as
+                             * some other thread may rely on the key_count data. Only do so if we
+                             * successfully committed.
+                             */
+                            cc.coll.increase_key_count(added_count);
+                        } else {
+                            added_count = 0;
+                        }
+                    }
+                }
+
+                /* Sleep the duration defined by the op_rate. */
+                tw->sleep();
+            }
+            /* Reset our cursor to avoid pinning content. */
+            testutil_check(cc.cursor->reset(cc.cursor.get()));
+            counter++;
+            if (counter == collections_per_thread)
+                counter = 0;
+            testutil_assert(counter < collections_per_thread);
+        }
+        /* Make sure the last transaction is rolled back now the work is finished. */
+        tw->txn.try_rollback();
+    }
+
+    void
+    validate(bool tracking_enabled, const std::string &operation_table_name, const std::string &,
```

Comment:
You can remove the `tracking_enabled` and `operation_table_name` from the function definition since they are not used.